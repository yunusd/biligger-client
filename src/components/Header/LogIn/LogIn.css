.logIn {
  padding: 10px 12px!important;
  width: 500px!important;
}

.logInRow {
  padding: 0px!important;
}

.logInLeft {
  background-color: #527ec0!important;
}

.logInRight {
  padding: 20px 10px!important;
}

.logInLeftText {
  color: white!important;
  height: 100%!important;
  display: flex!important;
  flex-direction: column!important;
  align-items: center!important;
  justify-content: center!important;
}