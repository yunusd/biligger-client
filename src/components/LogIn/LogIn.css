/* .login-grid {
  height: 100%!important;
} */

.login-column {
  max-width: 300px!important;
  margin-top: 50px!important;
}

.login-header {
  color: #527ec0!important;
  margin-top: 80px!important;
}

.login-segment {
  padding: 20px 40px!important;
  border: none!important;
}

.boxs {
	-webkit-box-shadow: 0px 10px 20px 0px rgba(50, 50, 50, 0.52);
	-moz-box-shadow:    0px 10px 20px 0px rgba(50, 50, 50, 0.52);
  box-shadow:         0px 10px 20px 0px rgba(50, 50, 50, 0.52);
  border-radius: 5px;  
}