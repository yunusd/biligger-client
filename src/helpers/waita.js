export default (amount = 0) => new Promise(resolve => setTimeout(resolve, amount));
