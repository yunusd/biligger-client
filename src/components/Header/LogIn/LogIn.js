import React, { Component } from 'react';
import axios from 'axios';

import {
  Button, Header, Form, Popup, Grid, Divider,
} from 'semantic-ui-react';
import './LogIn.css';
import { Link } from 'react-router-dom';

class LogIn extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit() {
    const { username, password } = this.state;
    axios.post('https://localhost:3000/auth', {
      username,
      password,
    }, { withCredentials: 'include' }).then((res) => {
      window.location.reload();
      return res;
    }).catch((error) => {
      const unitTestHandleSubmit = this.props.handleSubmit;
      if (unitTestHandleSubmit) {
        this.props.handleSubmit(error); // added for unit test
      }
      this.setState({ error: true });
      return error;
    });
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    const { error = '' } = this.state;
    const {
      button, location,
    } = this.props;

    const urlCheck = location ? location.pathname === '/giriş-yap' : false;
    const errorMessage = `${error && 'Kullanıcı adı ya da şifre yanlış!'}`;

    return (
      <div>
        { urlCheck
        ? <h5>GİRİŞ YAP</h5>
        : (
          <Popup
            trigger={button}
            on="click"
            flowing
            position="bottom right"
            className="logIn"
            content={(
              <Grid columns={2} divided>
                <Grid.Row className="logInRow" stretched>
                  <Grid.Column className="logInLeft">
                    <Header
                      content={errorMessage || 'Hoşgeldiniz'}
                      size="huge"
                      textAlign="center"
                      className="logInLeftText"
                    />
                  </Grid.Column>
                  <Grid.Column className="logInRight">
                    <Form onSubmit={this.handleSubmit}>
                      <Form.Input label="Kullanıcı adı" placeholder="Kullanıcı adı" name="username" onChange={this.handleChange} />
                      <Form.Input type="password" label="Şifre" placeholder="Şifre" name="password" onChange={this.handleChange} />
                      <Button
                        type="submit"
                        content="Giriş Yap"
                        floated="right"
                      />
                    </Form>

                    <Divider />

                    <div>
                      Henüz hesabınız yok mu?
                      <Link to="/kayıt"> Kayıt Ol</Link>
                    </div>
                  </Grid.Column>
                </Grid.Row>
              </Grid>
            )}
          />

        )
        }
      </div>
    );
  }
}

export default LogIn;
