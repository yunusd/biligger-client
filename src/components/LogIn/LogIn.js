import React from 'react';
import { useApolloClient } from 'react-apollo-hooks';
import { Helmet } from 'react-helmet';
import axios from 'axios';
import { Link, Redirect } from 'react-router-dom';
import {
  Button, Header, Form, Grid, Image, Segment, Divider, Message,
} from 'semantic-ui-react';
import { withFormik } from 'formik';
import * as Yup from 'yup';

import { GET_AUTH_STATUS } from '../../queries';
import logo from '../../logo.png';
import './LogIn.css';

const LogInSchema = Yup.object().shape({
  username: Yup.string()
    .required('Kullanıcı adı gerekli!'),
  password: Yup.string()
    .required('Şifre gerekli!'),
});

const LogInForm = (props) => {
  const client = useApolloClient();

  const { currentUser } = client.readQuery({ query: GET_AUTH_STATUS });
  if (currentUser.isLoggedIn) return <Redirect to="/" />;

  const {
    values,
    touched,
    errors,
    handleChange,
    handleSubmit,
    status,
    isSubmitting,
  } = props;

  const { username = '', isRedirect = '' } = props;
  const user = `${isRedirect && `Hoşgeldin, ${username}`}`;

  const errorExists = Object.getOwnPropertyNames(errors).length > 0
  && Object.getOwnPropertyNames(touched).length > 0;
  const serverValidationErrors = status.errors ? status.errors : null;

  return (
    <Grid textAlign="center" className="login-grid" verticalAlign="bottom">
      <Helmet>
        <title>Giriş Yap - Biligger</title>
      </Helmet>
      <Grid.Column className="login-column">
        <Image as={Link} to="/" src={logo} size="medium" />
        <Header as="h2" className="login-header" textAlign="center">
          {user || 'Hoşgeldiniz'}
        </Header>
        <div className="boxs">
          <Segment raised className="login-segment">
            {
              errorExists || !!serverValidationErrors
              ? (
                <Message
                  error
                  list={[
                    serverValidationErrors && serverValidationErrors.message,
                    errors.username,
                    errors.password,
                  ]}
                />
              )
              : null
            }
            <Form onSubmit={handleSubmit}>
              <Form.Input error={!!errors.username} type="text" placeholder="Kullanıcı adı" name="username" value={values.username} onChange={handleChange} />
              <Form.Input error={!!errors.password} type="password" placeholder="Şifre" name="password" value={values.password} onChange={handleChange} />
              <Button
                disabled={isSubmitting}
                type="submit"
                content="Giriş Yap"
              />
            </Form>
            <Divider />
            <Link to="/giris/sifirla" style={{ color: '#49ba6f' }}>Şifreni mi unuttun?</Link>
            <br />
            Henüz kayıt olmadın mı?
            <br />
            <Link to="/kayit" onClick={() => { if (props.location.pathname === 'kayit') props.history.replace('/kayit'); }} style={{ color: '#49ba6f' }}>Kayıt Ol</Link>
          </Segment>
        </div>
      </Grid.Column>
    </Grid>
    );
};

const LogIn = withFormik({
  mapPropsToValues: () => ({
    username: '',
    password: '',
  }),
  mapPropsToStatus: () => ({
    errors: null,
  }),
  validationSchema: LogInSchema,
  handleSubmit: async (values, { setSubmitting, setStatus, props }) => {
    const { username, password } = values;

    axios.post('/auth', {
      username,
      password,
    }, { withCredentials: 'include' }).then((res) => {
      window.location.replace('/');
      return res;
    }).catch((error) => {
      setSubmitting(false);
      const unitTestHandleSubmit = props.handleTestSubmit;
      if (unitTestHandleSubmit) {
        props.handleTestSubmit(error); // added for unit test
      }
      setStatus({
        errors: {
          message: 'Kullanıcı adı ya da Şifre yanlış.',
        },
      });
      return error;
    });
  },
  displayName: 'BasicForm',
})(LogInForm);


export default LogIn;
