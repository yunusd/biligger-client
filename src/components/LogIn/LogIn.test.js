import React from 'react';
import { shallow } from 'enzyme';
import {
  Header, Form, FormInput,
} from 'semantic-ui-react';

import waita from '../../helpers/waita';
import LogIn from './LogIn';

const fetchMock = require('fetch-mock');


describe('LogIn', () => {
  it('render without crashing', () => {
    const wrapper = shallow(<LogIn />).exists();
    expect(wrapper).toBeTruthy();
  });

  it('render correctly', () => {
    const wrapper = shallow(<LogIn />);
    expect(wrapper).toMatchSnapshot();
  });

  it('return username after successful login', () => {
    const wrapper = shallow(<LogIn username="Foo" isRedirect />);
    const login = wrapper.find(Header).dive().find('.huge').text();
    expect(login).toContain('Foo');
  });

  it('change state when input change', () => {
    const wrapper = shallow(<LogIn />);
    wrapper.find(FormInput).at(0).simulate('change', { target: { name: 'username', value: 'foo' } });
    wrapper.find(FormInput).at(1).simulate('change', { target: { name: 'password', value: 'bar' } });
    expect(wrapper.state()).toMatchObject({ username: 'foo', password: 'bar' });
  });

  it('should return token if credentials are valid', async () => {
    const payload = JSON.stringify({
      grant_type: 'password',
      username: 'foo',
      password: 'bar',
      scope: 'offline_access',
    });

    fetchMock.mock((url, opts) => (url === 'http://localhost:3000/oauth/token' && opts.body === payload), {
      access_token: 'dummy-token',
    }).catch({ error: 'not match' });

    const fakeEvent = { preventDefault: () => {} };
    const wrapper = shallow(<LogIn />);
    wrapper.find(FormInput).at(0).simulate('change', { target: { name: 'username', value: 'foo' } });
    wrapper.find(FormInput).at(1).simulate('change', { target: { name: 'password', value: 'bar' } });
    wrapper.find(Form).simulate('submit', fakeEvent);
    await waita(2);
    expect(localStorage.getItem('access_token')).toBe('dummy-token');
    fetchMock.restore();
  });

  it('should return an error if credentials are not valid', async () => {
    const payload = JSON.stringify({
      grant_type: 'password',
      username: 'foo',
      password: 'bar',
      scope: 'offline_access',
    });

    fetchMock.mock((url, opts) => (url === 'http://localhost:3000/oauth/token' && opts.body !== payload), {
      error: 'wrong credentials!',
    }).catch({});

    const handleSubmit = jest.fn(res => res);
    const fakeEvent = { preventDefault: () => {} };
    const wrapper = shallow(<LogIn handleSubmit={handleSubmit} />);
    wrapper.find(FormInput).at(0).simulate('change', { target: { name: 'username', value: 'foo2' } });
    wrapper.find(FormInput).at(1).simulate('change', { target: { name: 'password', value: 'bar' } });
    wrapper.find(Form).simulate('submit', fakeEvent);
    await waita(2);

    const error = handleSubmit.mock.results[0] ? handleSubmit.mock.results[0].value.error : false;
    /* manually set error to the state because enzyme doesn't re-render component
    and cause to not pass error message to the component */
    if (error) wrapper.setState({ error });

    expect(wrapper.find(Header).dive().text()).toContain('Hata :('); // HTML
    expect(error).toBe('wrong credentials!'); // Fetch response
    fetchMock.restore();
  });
});
