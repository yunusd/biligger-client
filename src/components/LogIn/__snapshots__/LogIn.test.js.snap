// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "centered": true,
      "children": <ModalContent
        className="modal-content"
        image={true}
      >
        <div
          className="login-left"
        >
          <Header
            size="huge"
          >
            GİRİŞ YAP
            <br />
            <pre />
          </Header>
        </div>
        <ModalDescription>
          <Form
            as="form"
            onSubmit={[Function]}
          >
            <FormInput
              as={[Function]}
              control={[Function]}
              label="Kullanıcı adı"
              name="username"
              onChange={[Function]}
              placeholder="Kullanıcı adı"
            />
            <FormInput
              as={[Function]}
              control={[Function]}
              label="Şifre"
              name="password"
              onChange={[Function]}
              placeholder="Şifre"
              type="password"
            />
            <Button
              as="button"
              content="Giriş Yap"
              floated="right"
              type="submit"
            />
          </Form>
        </ModalDescription>
      </ModalContent>,
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": "blurring",
      "eventPool": "Modal",
      "open": undefined,
      "trigger": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="login-left"
          >
            <Header
              size="huge"
            >
              GİRİŞ YAP
              <br />
              <pre />
            </Header>
          </div>,
          <ModalDescription>
            <Form
              as="form"
              onSubmit={[Function]}
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                label="Kullanıcı adı"
                name="username"
                onChange={[Function]}
                placeholder="Kullanıcı adı"
              />
              <FormInput
                as={[Function]}
                control={[Function]}
                label="Şifre"
                name="password"
                onChange={[Function]}
                placeholder="Şifre"
                type="password"
              />
              <Button
                as="button"
                content="Giriş Yap"
                floated="right"
                type="submit"
              />
            </Form>
          </ModalDescription>,
        ],
        "className": "modal-content",
        "image": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Header
              size="huge"
            >
              GİRİŞ YAP
              <br />
              <pre />
            </Header>,
            "className": "login-left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                "GİRİŞ YAP",
                <br />,
                <pre />,
              ],
              "size": "huge",
            },
            "ref": null,
            "rendered": Array [
              "GİRİŞ YAP",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "pre",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Form
              as="form"
              onSubmit={[Function]}
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                label="Kullanıcı adı"
                name="username"
                onChange={[Function]}
                placeholder="Kullanıcı adı"
              />
              <FormInput
                as={[Function]}
                control={[Function]}
                label="Şifre"
                name="password"
                onChange={[Function]}
                placeholder="Şifre"
                type="password"
              />
              <Button
                as="button"
                content="Giriş Yap"
                floated="right"
                type="submit"
              />
            </Form>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "form",
              "children": Array [
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  label="Kullanıcı adı"
                  name="username"
                  onChange={[Function]}
                  placeholder="Kullanıcı adı"
                />,
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  label="Şifre"
                  name="password"
                  onChange={[Function]}
                  placeholder="Şifre"
                  type="password"
                />,
                <Button
                  as="button"
                  content="Giriş Yap"
                  floated="right"
                  type="submit"
                />,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": [Function],
                  "control": [Function],
                  "label": "Kullanıcı adı",
                  "name": "username",
                  "onChange": [Function],
                  "placeholder": "Kullanıcı adı",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": [Function],
                  "control": [Function],
                  "label": "Şifre",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Şifre",
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "content": "Giriş Yap",
                  "floated": "right",
                  "type": "submit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "centered": true,
        "children": <ModalContent
          className="modal-content"
          image={true}
        >
          <div
            className="login-left"
          >
            <Header
              size="huge"
            >
              GİRİŞ YAP
              <br />
              <pre />
            </Header>
          </div>
          <ModalDescription>
            <Form
              as="form"
              onSubmit={[Function]}
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                label="Kullanıcı adı"
                name="username"
                onChange={[Function]}
                placeholder="Kullanıcı adı"
              />
              <FormInput
                as={[Function]}
                control={[Function]}
                label="Şifre"
                name="password"
                onChange={[Function]}
                placeholder="Şifre"
                type="password"
              />
              <Button
                as="button"
                content="Giriş Yap"
                floated="right"
                type="submit"
              />
            </Form>
          </ModalDescription>
        </ModalContent>,
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": "blurring",
        "eventPool": "Modal",
        "open": undefined,
        "trigger": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="login-left"
            >
              <Header
                size="huge"
              >
                GİRİŞ YAP
                <br />
                <pre />
              </Header>
            </div>,
            <ModalDescription>
              <Form
                as="form"
                onSubmit={[Function]}
              >
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  label="Kullanıcı adı"
                  name="username"
                  onChange={[Function]}
                  placeholder="Kullanıcı adı"
                />
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  label="Şifre"
                  name="password"
                  onChange={[Function]}
                  placeholder="Şifre"
                  type="password"
                />
                <Button
                  as="button"
                  content="Giriş Yap"
                  floated="right"
                  type="submit"
                />
              </Form>
            </ModalDescription>,
          ],
          "className": "modal-content",
          "image": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Header
                size="huge"
              >
                GİRİŞ YAP
                <br />
                <pre />
              </Header>,
              "className": "login-left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "GİRİŞ YAP",
                  <br />,
                  <pre />,
                ],
                "size": "huge",
              },
              "ref": null,
              "rendered": Array [
                "GİRİŞ YAP",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "pre",
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Form
                as="form"
                onSubmit={[Function]}
              >
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  label="Kullanıcı adı"
                  name="username"
                  onChange={[Function]}
                  placeholder="Kullanıcı adı"
                />
                <FormInput
                  as={[Function]}
                  control={[Function]}
                  label="Şifre"
                  name="password"
                  onChange={[Function]}
                  placeholder="Şifre"
                  type="password"
                />
                <Button
                  as="button"
                  content="Giriş Yap"
                  floated="right"
                  type="submit"
                />
              </Form>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "form",
                "children": Array [
                  <FormInput
                    as={[Function]}
                    control={[Function]}
                    label="Kullanıcı adı"
                    name="username"
                    onChange={[Function]}
                    placeholder="Kullanıcı adı"
                  />,
                  <FormInput
                    as={[Function]}
                    control={[Function]}
                    label="Şifre"
                    name="password"
                    onChange={[Function]}
                    placeholder="Şifre"
                    type="password"
                  />,
                  <Button
                    as="button"
                    content="Giriş Yap"
                    floated="right"
                    type="submit"
                  />,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": [Function],
                    "control": [Function],
                    "label": "Kullanıcı adı",
                    "name": "username",
                    "onChange": [Function],
                    "placeholder": "Kullanıcı adı",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": [Function],
                    "control": [Function],
                    "label": "Şifre",
                    "name": "password",
                    "onChange": [Function],
                    "placeholder": "Şifre",
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "button",
                    "content": "Giriş Yap",
                    "floated": "right",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
